lightning/src/events/mod.rs

```diff
@@ -774,10 +774,16 @@ pub enum Event {
 		/// Information for claiming this received payment, based on whether the purpose of the
 		/// payment is to pay an invoice or to send a spontaneous payment.
 		purpose: PaymentPurpose,
-		/// The `channel_id` indicating over which channel we received the payment.
-		via_channel_id: Option<ChannelId>,
-		/// The `user_channel_id` indicating over which channel we received the payment.
-		via_user_channel_id: Option<u128>,
+		/// The `channel_id`(s) indicating over which channel(s) we received the payment.
+		/// - In a non-MPP scenario, this will contain a single `channel_id` where the payment was received.
+		/// - In an MPP scenario, this will contain multiple `channel_id`s corresponding to the channels over
+		///   which the payment parts were received.
```

Comment:
IMO we can delete this, similar below. Could we document for which versions these vecs will be empty as well, if any?

Reply:
Done! [pr3655.03](https://github.com/shaavan/rust-lightning/commits/pr3655.03)
I have deleted the docs section and added a note that this will be empty vector for LDK version `<= 0.0.112`
Thanks for the pointer!